ocaml/typing/typecore.ml

```diff
@@ -5508,20 +5521,33 @@ and type_expect_
                                (Pexp_constant (Pconst_string(name,_,None)));
                              pexp_loc = name_loc;
                              _ }
-                          , [Nolabel, arg]))
+                          , ([Nolabel, arg]|
+                             [Labelled "enabled_at_init",
+                              { pexp_desc =
+                                  Pexp_construct({ txt = Longident.Lident "false"; _ },
+                                                 None); _ };
+                              Nolabel, arg])))
                    ; _ }
                   , _)}]) ->
-        check_probe_name name name_loc env;
-        let env = Env.add_lock Alloc_mode.global env in
-        Env.add_probe name;
-        let exp = type_expect env mode_global arg
-                    (mk_expected Predef.type_unit) in
-        rue {
-          exp_desc = Texp_probe {name; handler=exp};
-          exp_loc = loc; exp_extra = [];
-          exp_type = instance Predef.type_unit;
-          exp_attributes = sexp.pexp_attributes;
-          exp_env = env }
+        mk_probe ~enabled_at_init:false name name_loc arg
+      | PStr
+          ([{ pstr_desc =
+                Pstr_eval
+                  ({ pexp_desc =
+                       (Pexp_apply
+                          ({ pexp_desc=
+                               (Pexp_constant (Pconst_string(name,_,None)));
+                             pexp_loc = name_loc;
+                             _ }
+                          , [Labelled "enabled_at_init",
+                             { pexp_desc =
+                                 Pexp_construct({ txt = Longident.Lident "true"; _ },
+                                                None);
+                               _ };
+                             Nolabel, arg]))
+                   ; _ }
+                  , _)}]) ->
```

Comment:
This match is getting a little wild.  Maybe factor out the common bits of the two cases, like:

```ocaml
let (name, name_loc, args) = 
  begin match payload with
  | PStr
      ([{ pstr_desc =
            Pstr_eval
              ({ pexp_desc =
                   (Pexp_apply
                      ({ pexp_desc=
                           (Pexp_constant (Pconst_string(name,_,None)));
                         pexp_loc = name_loc;
                         _ }
                      , args))})}]) -> name, name_loc, args
  | raise (Error (loc, env, Probe_format))
in
let (arg, enabled_at_init) =
  match args with
  ...
in
mk_probe ~enabled_at_init name name_loc arg
```

Reply:
It might seem a little unclear to a user whether `[%probe.enable_at_startup ...]` is a new probe or some kind of command modifying an existing probe of that name.  But I don't feel strongly either way - it's fine if you two like it.

I don't think the compiler interprets the dots in extension names at all.  In places where both `ext_name` and `ocaml.ext_name` work, I think it's because we're just explicitly checking for both (see all the locals stuff in typecore.ml, for example).  I don't know much about ppxlib.  I didn't have to change it when adding attributes for things like include functor, so I'm guessing it just ignores attributes it doesn't know about, and maybe the same is true for extensions?

Reply:
I slightly prefer ``[%probe.enable_at_startup ...]`` since the modifier changes how the probe is emitted rather than what functionality it wraps. Perhaps naming it ``[%probe.default_enabled ...]`` is clearer?

Reply:
After some offline bike-shedding, we keep the syntax in the current patch.